app/renderer/bookmarks.js

```diff
@@ -0,0 +1,128 @@
+/*
+* Vieb - Vim Inspired Electron Browser
+* Copyright (C) 2019-2022 Jelmer van Arnhem
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+"use strict"
+const {
+    writeJSON,
+    readJSON,
+    joinPath,
+    notify,
+    isAbsolutePath,
+    appData
+} = require("../util")
+const {
+    currentPage,
+    getSetting,
+    currentTab
+} = require("./common")
+
+let bookmarks = {}
+let bookmarksFile = ""
+
+const validOptions = [
+    "bg",
+    "fg",
+    "keywords",
+    "name",
+    "path",
+    "tag",
+    "title",
+    "url"
+]
+
+const setBookmarkSettings = () => {
+    const setFile = getSetting("bookmarksfile")
+    if (setFile === "bookmarks") {
+        bookmarksFile = joinPath(appData(), "bookmarks")
+    } else if (isAbsolutePath(setFile)) {
+        bookmarksFile = setFile
+    } else {
+        bookmarksFile = joinPath(appData(), setFile)
+    }
+    bookmarks = readJSON(bookmarksFile)
+}
+
+const addBookmark = input => {
+    const newbookmark = {}
+    // If there's "=" or "~" we parse the input.
+    if (input.join().includes("=") || input.join().includes("~")) {
+        // Retain spaces for multiple word inputs.
+        const options = input.join(" ").split("~")
+        for (let i = 0; i < options.length; i++) {
+            // Get key and value: [0,1]
+            const keyAndValue = options[i].split("=")
+            const allValue = keyAndValue.slice(1).join("")
+            if (allValue !== "") {
+                newbookmark[keyAndValue[0]] = allValue
+            }
+        }
+    } else {
+        // If there's only text, we take it as name for current page bookmark
+        newbookmark.name = input.join(" ")
+        if (newbookmark.name === "") {
+            // When user doesn't enter anything delete empty string
+            delete newbookmark.name
+        }
+    }
+    // Fill missing essential data from relevant sources
+    if (typeof newbookmark.name === "undefined") {
+        newbookmark.name = currentTab().querySelector("span").textContent
+    }
+    if (typeof newbookmark.title === "undefined") {
+        newbookmark.title = currentTab().querySelector("span").textContent
+    }
+    if (typeof newbookmark.url === "undefined") {
+        newbookmark.url = currentPage().src
+    }
```

Comment:
Can you make all string comparisons here rely on the falsyness/truthyness of empty/filled strings? as in:
```js
    if (!newbookmark.name?.trim()) {
        newbookmark.name = currentTab().querySelector("span").textContent
    }
```